crates/oxc_linter/src/rules/unicorn/no_instanceof_array.rs

```diff
@@ -0,0 +1,107 @@
+use oxc_ast::ast::Expression;
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_formatter::Gen;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use oxc_syntax::operator::BinaryOperator;
+
+use crate::{context::LintContext, fixer::Fix, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Use `Array.isArray()` instead of `instanceof Array`.")]
+#[diagnostic(severity(warning))]
+struct NoInstanceofArrayDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoInstanceofArray;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Require `Array.isArray()` instead of `instanceof Array`.
+    ///
+    /// ### Why is this bad?
+    /// The instanceof Array check doesn't work across realms/contexts, for example, frames/windows in browsers or the vm module in Node.js.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// array instanceof Array;
+    /// [1,2,3] instanceof Array;
+    /// ```
+    NoInstanceofArray,
+    correctness
```

Comment:
I have a feeling this will blow up some codebases, we'll see :-)

Reply:
You mean we have to change it to `restriction` instead?

Reply:
I'll change it if it blows up when I do some testing on real codebases before release.